libs/langchain/langchain/chat_models/anthropic.py

```diff
@@ -18,46 +14,29 @@
     HumanMessage,
     SystemMessage,
 )
-from langchain.schema.output import ChatGenerationChunk
+from langchain.schema.output import ChatGeneration, ChatGenerationChunk, ChatResult
 
 
-class ChatAnthropic(BaseChatModel, _AnthropicCommon):
-    """Anthropic's large language chat model.
+class ChatAnthropicMessageConverter:
```

Comment:
What do you guys think about making this class into a simple function (see below). For me at least it just reads a bit easier and a class name called "Converter" may be is indicative that it should just be a function?

```python
def convert_messages_to_prompt_anthropic(
    messages: List[BaseMessage],
    human_prompt: str = "\n\nHuman:",
    ai_prompt: str = "\n\nAssistant:"
) -> str:
   """Format the prompts in the expected format for the Anthropic model"""
   
    formatted_messages = []
    for message in messages:
        if isinstance(message, ChatMessage):
            formatted_messages.append(f"\n\n{message.role.capitalize()}: {message.content}")
        elif isinstance(message, HumanMessage):
            formatted_messages.append(f"{human_prompt} {message.content}")
        elif isinstance(message, AIMessage):
            formatted_messages.append(f"{ai_prompt} {message.content}")
        elif isinstance(message, SystemMessage):
            formatted_messages.append(f"{human_prompt} <admin>{message.content}</admin>")
        else:
            raise ValueError(f"Got unknown type {message}")
    
    if not isinstance(messages[-1], AIMessage):
        formatted_messages.append(f"{ai_prompt} ")
    
    formatted_prompt = "".join(formatted_messages).rstrip()
    return formatted_prompt
```

Reply:
I like this approach and refactored the code with a similar implementation.